### PR TITLE
Adding Support for Debian 10 and fixing Docker packages

### PR DESCRIPTION
**Pull request by daviddjh**
Added Debian 10 to the list of platforms in metadata and spec_helper. Made sure platforms got the right docker packages.

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -6,15 +6,26 @@
     "#{default['osl-docker']['package']['version']}-#{default['osl-docker']['package_release']}"
   default['osl-docker']['package']['package_name'] = 'docker-ce'
   default['osl-docker']['package_cli_name'] = 'docker-ce-cli'
+  default['osl-docker']['package']['setup_docker_repo'] = true
+  default['osl-docker']['package']['setup_docker_repo'] = false if node['platform_version'].to_i >= 8
```
Comment (ramereth) [inline attributes/default.rb]: This doesn't look right. why is this changing?

File: attributes/default.rb [hunk 2]
```diff
@@ -6,15 +6,26 @@
...
+  when 9
+    default['osl-docker']['package']['package_name'] = 'docker-ce'
+    default['osl-docker']['package_cli_name'] = 'docker-ce-cli'
+    default['osl-docker']['package']['version'] = '5:18.09.2'
+    default['osl-docker']['package']['package_version'] = '5:18.09.2~3-0~debian-stretch'
+  when 10
+    default['osl-docker']['package']['package_name'] = 'docker-ce'
+    default['osl-docker']['package_cli_name'] = 'docker-ce-cli'
+    default['osl-docker']['package']['version'] = '5:18.09.2'
+    default['osl-docker']['package']['package_version'] = '5:18.09.2~3-0~debian-buster'
```
Comment (ramereth) [inline attributes/default.rb]: I think you can remove the case statement and got back to what we had before but change it to do this so it works with any version:
```suggestion
    default['osl-docker']['package']['package_version'] = "5:18.09.2~3-0~debian-#{node['lsb']['codename']}"
```

File: attributes/default.rb [hunk 3]
```diff
@@ -6,15 +6,26 @@
...
+  end
+
 end
 case node['kernel']['machine']
-when 'x86_64'
-  default['osl-docker']['package']['setup_docker_repo'] = false if node['platform_version'].to_i >= 8
```
Comment (ramereth) [inline attributes/default.rb]: Why was this touched?

File: recipes/default.rb [hunk 4]
```diff
@@ -11,10 +11,12 @@
 #     http://www.apache.org/licenses/LICENSE-2.0
 #
 # Unless required by applicable law or agreed to in writing, software
+
```
Comment (ramereth) [inline recipes/default.rb]: Remove this newline please.

File: recipes/default.rb [hunk 5]
```diff
@@ -30,17 +32,17 @@
     end
   end
 
-  yum_version_lock node['osl-docker']['package_cli_name'] do
+  yum_version_lock node['osl-docker']['package']['package_name'] do
```
Comment (ramereth) [inline recipes/default.rb]: Why was this changed?

**Comment by ramereth** [comment]
[Upgrade Workstations to Debian 10](https://trello.com/c/FNOgnBLf/1394-upgrade-workstations-to-debian-10)

File: recipes/default.rb (hunk 5 above)
Comment (daviddjh) [reply recipes/default.rb]: Cent OS 7 refused to version lock the cli package if they were in the other order for some odd reason.

File: attributes/default.rb (hunk 3 above)
Comment (daviddjh) [reply attributes/default.rb]: It set that variable to false for debian 9 and 10 then they wouldn't converge. I think it was only ment for centos 8.

File: recipes/default.rb (hunk 5 above)
Comment (ramereth) [reply recipes/default.rb]: Strange, but I confirm it as well.
